### DISCUSSION TITLE
Should I remove all memo, useMemo and useCallback instances from my codebase if using react-compiler?

### DISCUSSION THREAD

**Discussion by TowhidKashem** [reactwg/react-compiler#39] [General Questions] [answered]
Should I remove all current usage of memo(), useMemo() and useCallback() from my codebase if I enable react-compiler globally on my project?

Or can I leave them in, and react compiler will just ignore my usages of these methods and do it's own optimizations and use that in the final build?

**Comment by poteto** [comment, marked as answer]
Previously answered here: https://github.com/reactwg/react-compiler/discussions/16#discussioncomment-9991176

**Reply by TowhidKashem** [reply]
Thank you, this answers my question but I have another related one. You said to not do manual memoization once react compiler is enabaled for new features but when I do that I get eslint warnings and errors like these:

```
The 'handleGoToApp' function makes the dependencies of useEffect Hook (at line 177) change on every render. To fix this, wrap the definition of 'handleGoToApp' in its own useCallback() Hook.eslint[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)
```
Should we be turning off all these eslint rules once react-compiler is enabled? I should mention I have the `react-compiler/react-compile` package and rules enabled on this project but still get these errors.
